core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -132,6 +136,12 @@ class GameInfo {
     fun getAliveCityStates() = civilizations.filter { it.isAlive() && it.isCityState() }
     fun getAliveMajorCivs() = civilizations.filter { it.isAlive() && it.isMajorCiv() }
 
+    fun getGameSpeed() = gameSpeedObject
+    fun setGameSpeed(newGameSpeed: GameSpeed) {
+        gameSpeedObject = newGameSpeed
+        gameParameters.gameSpeed = newGameSpeed.name
+    }
+
```

Comment:
I realized that the custom setter isn't even needed since starting a new game (or loading a saved one) goes something like
1. set `gameParameters.speed` using dropdown menu (or read from file)
2. set `speed` object using `ruleSet.speeds[gameParameters.speed]`